src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -453,6 +453,7 @@ class Peer(
                 .combine(currentTipFlow.filterNotNull()) { walletState, currentTip -> Pair(walletState, currentTip.first) }
                 .combine(swapInFeeratesFlow.filterNotNull()) { (walletState, currentTip), feerate -> Triple(walletState, currentTip, feerate) }
                 .combine(nodeParams.liquidityPolicy) { (walletState, currentTip, feerate), policy -> TrySwapInFlow(currentTip, walletState, feerate, policy) }
+                .combine(channelsFlow.filter { it.values.all { channel -> channel !is Offline && channel !is Syncing } }) { req, _ -> req } // this is only for synchronization, we discard the channels data
```

Comment:
NB: this is a simplification of `waitForPeerReady()`, which doesn't handles the "local syncing" corner case.

Reply:
If you look at the handler for `RequestChannelOpen`, in the "local syncing" case requests always end up being ignored. It's probably a (rare) bug, but technically not a regression 😓